packages/contracts/contracts/UnirepApp.sol

```diff
@@ -370,32 +389,42 @@ contract UnirepApp is Ownable {
     /**
      * Claim the daily login reputation
      * @param publicSignals: public signals
-     * @param proof: epoch key proof
+     * @param proof: daily claim proof
      */
     function claimDailyLoginRep(
         uint256[] calldata publicSignals,
-        uint256[8] calldata proof
+        uint256[8] calldata proof,
+        bytes32 identifier
     ) public onlyOwner() {
+        _updateDailyEpochIfNeeded();
+
+        DailyClaimVHelper dailyClaimVHelpers = DailyClaimVHelper(verifierHelperManager.registeredVHelpers(identifier));
+        DailyClaimVHelper.DailyClaimSignals memory signals = dailyClaimVHelpers.decodeDailyClaimSignals(publicSignals);
+
         // check if proof is used before
-        bytes32 nullifier = keccak256(abi.encodePacked(publicSignals, proof));
+        bytes32 nullifier = bytes32(signals.dailyNullifier);
         if (proofNullifier[nullifier]) {
             revert ProofHasUsed();
         }
 
         proofNullifier[nullifier] = true;
 
-        EpochKeyVerifierHelper.EpochKeySignals
-            memory signals = epkHelper.decodeEpochKeySignals(
-                publicSignals
-            );
-
         // check the epoch != current epoch (ppl can only post in current aepoch)
         uint48 epoch = unirep.attesterCurrentEpoch(signals.attesterId);
         if (signals.epoch > epoch) {
             revert InvalidEpoch();
         }
 
-        epkHelper.verifyAndCheckCaller(publicSignals, proof);
+        uint48 dailyEpoch = dailyEpochData.currentEpoch;
+        if (signals.dailyEpoch != dailyEpoch) {
+            revert InvalidDailyEpoch();
+        }
+
+        dailyClaimVHelpers.verifyAndCheck(publicSignals, proof);
+
+        if (signals.maxRep <= signals.minRep) {
+            revert NonNegetativeReputation();
+        }
```

Comment:
I found that there is a bug in the circuit
To prove the reputation is negative
you should prove `data[1] > data[0]`
or you can choose to prove `maxRep > 0` and `proveMaxRep > 0`
`maxRep` means `data[1] - data[0] >= maxRep`
See https://github.com/Unirep/Unirep/blob/36e5adf8c1eaa7398716ff3653327d5bd98181b3/packages/circuits/circuits/reputation.circom#L146